src/Pages/MainPage/Containers/MainContainer.tsx

```diff
@@ -52,18 +54,48 @@ const MainContainer: React.FC = () => {
     setEditingTaskId(null);
   };
 
-  const sortedTasks = tasks.slice().sort((a, b) => {
-    if (a.completed && !b.completed) return 1;
-    if (!a.completed && b.completed) return -1;
-    return 0;
-  });
+  const applyFilter = (tasks: Task[], filter: string): Task[] => {
+    switch (filter) {
+      case 'all':
+        return tasks.sort((a, b) => {
+          if (a.completed && !b.completed) return 1;
+          if (!a.completed && b.completed) return -1;
+          return 0;
+        });
+      case 'priority-high-low':
+        return tasks
+          .filter((task) => !task.completed)
+          .sort((a, b) => a.priority - b.priority);
+      case 'priority-low-high':
+        return tasks
+          .filter((task) => !task.completed)
+          .sort((a, b) => b.priority - a.priority);
+      case 'due-date-asc':
+        return tasks
+          .filter((task) => !task.completed)
+          .sort((a, b) => a.dueDate.getTime() - b.dueDate.getTime());
+      case 'completed':
+        return tasks.filter((task) => task.completed);
+      case 'active':
+        return tasks.filter((task) => !task.completed);
+      default:
+        return tasks;
+    }
+  };
+  
+  const filteredTasks = applyFilter(tasks, filter);
 
   return (
     <>
       <h1 className="App">To Do Application</h1>
-      <TaskForm addTask={addTask} deleteCompletedTasks={deleteCompletedTasks} />
+      <TaskForm 
+      addTask={addTask} 
+      deleteCompletedTasks={deleteCompletedTasks}
+      setFilter={setFilter}
+      filter={filter}
+      />
```

Comment:
A button to open up the form modal makes more sense from a UX perspective.
 
Can make a ticket to handle that later.

Reply:
Did not get this, should the filter be a modal instead of rendering as part of the form?

Reply:
There can be an Add task button which opens up a modal containing the add task form rather than it being open by default.

The filter button should be visible by default and not part of the modal.

Reply:
Oh noted, this looked cleaner and easier to implement for me, hence did not want to introduce a modal for the form. Can do that after finishing the app